apps/payroll/contracts/Payroll.sol

```diff
@@ -0,0 +1,586 @@
+pragma solidity 0.4.15;
+
+import "./oracle/OracleInterface.sol";
+
+import "@aragon/core/contracts/apps/App.sol";
+import "@aragon/core/contracts/common/Initializable.sol";
+import "@aragon/core/contracts/common/EtherToken.sol";
+import "@aragon/core/contracts/common/erc677/ERC677Receiver.sol";
+
+import "@aragon/core/contracts/zeppelin/token/ERC20.sol";
+import "@aragon/core/contracts/zeppelin/math/SafeMath.sol";
+
+import "@aragon/apps-finance/contracts/Finance.sol";
+//import "@aragon/apps-vault/contracts/Vault.sol";
+
+
+/**
+ * @title Payroll in multiple currencies
+ * @notice For the sake of simplicity lets asume USD is a ERC20 token
+ * Also lets asume we can 100% trust the exchange rate oracle
+ */
+contract Payroll is App, Initializable, ERC677Receiver {
+    using SafeMath for uint;
+
+    bytes32 constant public OWNER_WRITE_ROLE = bytes32(1);
+    bytes32 constant public OWNER_READ_ROLE = bytes32(2);
+    bytes32 constant public EMPLOYEE_ROLE = bytes32(3);
+    bytes32 constant public ORACLE_ROLE = bytes32(4);
+
+    struct Employee {
+        address accountAddress; // unique, but can be changed over time
+        mapping(address => bool) allowedTokens; // Could it be defined globally?
+        address[] allowedTokensArray;
+        mapping(address => uint256) allocation;
+        uint256 yearlyUSDSalary;
+        uint lastAllocation;
+        uint lastPayroll;
+        string name;
+    }
+
+    uint private numEmployees;
+    uint private nextEmployee; // starts at 1
+    mapping(uint => Employee) private employees;
+    mapping(address => uint) private employeeIds;
+    uint256 private yearlyTotalPayroll;
+
+    Finance public finance;
+    address public oracle;
+    address public usdToken;
+    EtherToken public etherToken;
+    mapping(address => uint256) private exchangeRates;
+    mapping(address => bool) globalTokens;
+    address[] globalTokensArray;
+
+    address public employer;
```

Comment:
Why is the 'employer' needed after this contract has been transitioned into an Aragon app?

Reply:
It was used in escapeHatch functions... which is also a mistake: funds should be sent to Finance which in turn would send them to the Vault. Fixing it.

Reply:
👍